# Increase buffer size in function pair_SAM_alignments_with_buffer to avoid this error:

# Error occured when processing SAM input (record #43311736 in file RIN10A_polyA.sorted.bam):
#  Maximum alignment buffer size exceeded while pairing SAM alignments.
#  [Exception type: ValueError, raised in __init__.py:671]

# reference: fanli user in seqanswers forum: http://seqanswers.com/forums/showpost.php?p=182624&postcount=18

diff -ru HTSeq-0.6.1p1.orig/HTSeq/__init__.py HTSeq-0.6.1p1/HTSeq/__init__.py
--- HTSeq-0.6.1p1.orig/HTSeq/__init__.py	2014-02-27 17:42:28.000000000 +0100
+++ HTSeq-0.6.1p1/HTSeq/__init__.py	2016-03-07 17:30:19.894201655 +0100
@@ -622,7 +622,7 @@
       warnings.warn( "%d reads with missing mate encountered." % mate_missing_count[0] )
 
 
-def pair_SAM_alignments_with_buffer( alignments, max_buffer_size=3000000 ):
+def pair_SAM_alignments_with_buffer( alignments, max_buffer_size=30000000 ):
 
    almnt_buffer = {}
    ambiguous_pairing_counter = 0
